lib/class-wp-theme-json-gutenberg.php

```diff
@@ -745,10 +745,10 @@ public function __construct( $theme_json = array( 'version' => WP_Theme_JSON_Gut
 		}
 
 		$this->theme_json    = WP_Theme_JSON_Schema_Gutenberg::migrate( $theme_json, $origin );
-		$registry            = WP_Block_Type_Registry::get_instance();
-		$valid_block_names   = array_keys( $registry->get_all_registered() );
+		$blocks_metadata     = static::get_blocks_metadata();
+		$valid_block_names   = array_keys( $blocks_metadata );
```

Comment:
I'm tentatively backporting this change too.

Core uses `::get_blocks_metadata()` to get valid block names. See: https://github.com/mukeshpanchal27/wordpress-develop/blob/trunk/src/wp-includes/class-wp-theme-json.php#L762

It has done it this way [since 5.9](https://github.com/wordpress/wordpress-develop/commit/60725c46ca1f657bb05ca16ea64d8b3476b8cbbb#diff-e12c3008d747d1bec5be9927c5e7b1ced0a88641abe52c278d495da936714817R335) 😄 

Then I found this Gutenberg PR from 2022, which doesn't appear to have been [backported](https://github.com/WordPress/gutenberg/pull/44658#issuecomment-1354425780):

- https://github.com/WordPress/gutenberg/pull/44658


Which is correct? 

I think maybe we could use what Core has since `::get_valid_block_style_variations()` will now call `::get_blocks_metadata()` anyway.

cc @ajlende @oandregal @aaronrobertshaw @mukeshpanchal27 for advice.

Reply:
Thanks for highlighting this. It might have been a factor in why the valid variations function evolved as it did.

> I think maybe we could use what Core has since ::get_valid_block_style_variations() will now call ::get_blocks_metadata() anyway.

Agreed. It saves another update in core too.